### PR TITLE
284 document improvement for `src/exit/payment`

### PR DESCRIPTION
**Pull request by boolafish**
### Note
- scan through all files in `src/exit/payment` and see if there is space for adding docuementation
- remove `IPaymentSPendingCondition` as it's leftover

ref: https://github.com/omisego/plasma-contracts/issues/284

### CODE REVIEW

File: plasma_framework/contracts/src/exits/payment/routers/PaymentInFlightExitRouter.sol [hunk 1]
```diff
@@ -156,6 +161,12 @@ contract PaymentInFlightExitRouter is IExitProcessor, Operated, OnlyWithValue {
         challengeCanonicityController.challenge(inFlightExitMap, args);
     }
 
+    /**
+     * @notice Respond to the non canonical challenge by providing position and prove the correctness of it.
+     * @param inFlightTx the rlp encoded in-flight transaction.
+     * @param inFlightTxPos the UTXO position of the in-flight exit. Should hardcode 0 for the outputIndex.
```
Comment (kevsul) [inline plasma_framework/contracts/src/exits/payment/routers/PaymentInFlightExitRouter.sol]:
```suggestion
     * @param inFlightTxPos the UTXO position of the in-flight exit. The outputIndex should be set to 0.
```

File: plasma_framework/contracts/src/exits/payment/routers/PaymentStandardExitRouter.sol [hunk 2]
```diff
@@ -63,8 +65,12 @@ contract PaymentStandardExitRouter is
         startStandardExitBond = BondSize.buildParams(INITIAL_BOND_SIZE, BOND_LOWER_BOUND_DIVISOR, BOND_UPPER_BOUND_MULTIPLIER);
     }
 
-    function standardExits(uint192 _exitId) public view returns (PaymentExitDataModel.StandardExit memory) {
-        return standardExitMap.exits[_exitId];
+    /**
+     * @notice Getter functions to recieve standard exit data of the PaymentExitGame.
```
Comment (kevsul) [inline plasma_framework/contracts/src/exits/payment/routers/PaymentStandardExitRouter.sol]:
```suggestion
     * @notice Getter functions to retrieve standard exit data of the PaymentExitGame.
```

File: plasma_framework/contracts/src/exits/payment/PaymentExitDataModel.sol [hunk 3]
```diff
@@ -1,9 +1,21 @@
 pragma solidity ^0.5.0;
 
+/**
```
Comment (thec00n) [inline plasma_framework/contracts/src/exits/payment/PaymentExitDataModel.sol]: Could we add a sample data for `StandardExit` and `InFlightExit` in json format so the context can be better understood?

File: plasma_framework/contracts/src/exits/payment/PaymentExitGame.sol [hunk 4]
```diff
@@ -11,6 +11,9 @@ import "../../vaults/Erc20Vault.sol";
 import "../interfaces/IStateTransitionVerifier.sol";
 import "../../utils/OnlyFromAddress.sol";
```
Comment (thec00n) [inline plasma_framework/contracts/src/exits/payment/PaymentExitGame.sol]: https://github.com/omisego/plasma-contracts/blob/e5e9e3d0f70b47e8483ee37cc464ce91da593789/plasma_framework/contracts/src/exits/payment/PaymentExitGame.sol#L53 > Don't what that means, could we rephrase this.

File: plasma_framework/contracts/src/exits/payment/PaymentTransactionStateTransitionVerifier.sol [hunk 5]
```diff
@@ -8,10 +8,14 @@ import "../../transactions/WireTransaction.sol";
 import "../../transactions/PaymentTransactionModel.sol";
 import "../../transactions/outputs/PaymentOutputModel.sol";
```
Comment (thec00n) [inline plasma_framework/contracts/src/exits/payment/PaymentTransactionStateTransitionVerifier.sol]: https://github.com/omisego/plasma-contracts/blob/406bed9da1316f8d75da31d458e0306467a3911c/plasma_framework/contracts/src/exits/payment/PaymentTransactionStateTransitionVerifier.sol#L32 could we remove the todo and open an issue for this on our gh if it does not exist already?

File: plasma_framework/contracts/src/exits/payment/PaymentExitGame.sol (hunk 4 above)
Comment (boolafish) [reply plasma_framework/contracts/src/exits/payment/PaymentExitGame.sol]: oh aw, I would remove this line instead then. I was planning to change the word from `token` to `ercContract` previously in the interface of `IExitProcessor`. This is basically trying the explain how `ercContract == token` here. However, as we decide to stick with current word usage, I think this comment is actually not needed anymore.

see: https://github.com/omisego/plasma-contracts/pull/285/files and https://github.com/omisego/plasma-contracts/pull/277

File: plasma_framework/contracts/src/exits/payment/PaymentTransactionStateTransitionVerifier.sol (hunk 5 above)
Comment (boolafish) [reply plasma_framework/contracts/src/exits/payment/PaymentTransactionStateTransitionVerifier.sol]: callout~ @pgebal can you help me understand what's the original issue on this? I think I used to understand what it means but not now ><

File: plasma_framework/contracts/src/exits/payment/PaymentExitDataModel.sol (hunk 3 above)
Comment (boolafish) [reply plasma_framework/contracts/src/exits/payment/PaymentExitDataModel.sol]: hmm sample data of InFlightExit would be pretty long though, what is the main part you'd like to have extra clarity?

```
{ exitStartTimestamp: 1569561351,
  exitMap: 80,
  position: 1000000000000,
  bondOwner: '0xf17f52151EbEF6C7334FAD080c5704D77216b732',
  bondSize: '37000000000000000',
  oldestCompetitorPosition: 0,
  inputs:
   [ { outputId:
        '0x28e655b6039b1b0b6b0bb2d9e73688248f5982090b328b1eb96d8f159b38a594',
       exitTarget: '0xC5fdf4076b8F3A5357c5E395ab970B5B54098Fef',
       token: '0x0000000000000000000000000000000000000000',
       amount: 667,
       piggybackBondSize: '28000000000000000' },
     { outputId:
        '0x550709e8d5a1bb24d69245e0d566252921907fcfbb233e0c29055373faff48d9',
       exitTarget: '0x821aEa9a577a9b44299B9c15c88cf3087F3b5544',
       token: '0x0000000000000000000000000000000000000000',
       amount: 667,
       piggybackBondSize: '28000000000000000' },
     { outputId:
        '0xd86a4212a1ff82b79ef3aedf737f714eef1ced10e8a014dcd60540abfe0d90d7',
       exitTarget: '0x0d1d4e623D10F9FBA5Db95830F7d3839406C6AF2',
       token: '0xFB88dE099e13c3ED21F80a7a1E49f8CAEcF10df6',
       amount: 667,
       piggybackBondSize: '28000000000000000' },
     { outputId:
        '0x088437a6f5532b5c550574551f6230150ac496af81eb27938bef5a5802be4a73',
       exitTarget: '0x0000000000000000000000000000000000000000',
       token: '0x0000000000000000000000000000000000000000',
       amount: 0,
       piggybackBondSize: 0 } ],
  outputs:
   [ { outputId:
        '0x2e6f452f1c2dfc17151c92a5b017f1cd6f64e2ea5aea7db74c6adea23b30de32',
       exitTarget: '0x2932b7A2355D6fecc4b5c0B6BD44cC31df247a2e',
       token: '0x0000000000000000000000000000000000000000',
       amount: 666,
       piggybackBondSize: '28000000000000000' },
     { outputId:
        '0xfda370418469bd46ae3f6cfce2cf66cf737e538c9e493824ce5fafa0fa3e10be',
       exitTarget: '0x2191eF87E392377ec08E7c08Eb105Ef5448eCED5',
       token: '0x0000000000000000000000000000000000000000',
       amount: 666,
       piggybackBondSize: '28000000000000000' },
     { outputId:
        '0xe4970c590baf7b5afcdd09f1b061230696edd2a9ec1220437d9fd628be4285f8',
       exitTarget: '0x0F4F2Ac550A1b4e2280d04c21cEa7EBD822934b5',
       token: '0xFB88dE099e13c3ED21F80a7a1E49f8CAEcF10df6',
       amount: 666,
       piggybackBondSize: '28000000000000000' },
     { outputId:
        '0x088437a6f5532b5c550574551f6230150ac496af81eb27938bef5a5802be4a73',
       exitTarget: '0x0000000000000000000000000000000000000000',
       token: '0x0000000000000000000000000000000000000000',
       amount: 0,
       piggybackBondSize: 0 } ] }
```

Comment (thec00n) [reply plasma_framework/contracts/src/exits/payment/PaymentExitDataModel.sol]: Never mind, I can generate what I am looking for with omg-js.

File: plasma_framework/contracts/src/exits/payment/PaymentTransactionStateTransitionVerifier.sol (hunk 5 above)
Comment (pgebal) [reply plasma_framework/contracts/src/exits/payment/PaymentTransactionStateTransitionVerifier.sol]: @boolafish  @thec00n the issue is that bytes array from calldata can not be copied to memory. So I had to consume inputTxs in the function body, couldn't pass it as argument.

**Comment by boolafish** [comment]
Seems like it is only the gas reporter that is failing.....
I've submit an issue to `eth-gas-reporter`: https://github.com/cgewecke/eth-gas-reporter/issues/184

Probably would wait a bit and if it still fails let's disable it temporary @@

**Comment by cgewecke** [comment]
Hi @boolafish, I'm visiting from eth-gas-reporter. Are the dependencies cached? If so would it be possible to retry this run after bumping the deps to their latest within the semver range?

**Comment by cgewecke** [comment]
@boolafish Oh actually - let me push a patch first very quickly. I will ping here within the hour when it's ready.

**Comment by cgewecke** [comment]
`0.2.12`?

**Comment by boolafish** [comment]
@cgewecke thanks! it seems working :)

**Comment by cgewecke** [comment]
@boolafish The diff you see here seems more or less correct?  Looks like there should no difference with this change-set.

I believe the `64` gas diff is something strange about ganache or the VM. I see that flicker occasionally on my own tests and am not sure why it's happening.

**Pull request merged**